### ISSUE TITLE
Suggestion: Add bookmarking or something similar

### ISSUE THREAD

**Issue by chozandrias76** [Chicken-Bones/NotEnoughItems#213]
Sometimes it is really annoying to have to find a block visually when you need to look at an entire mod's library of blocks. Would be really nice to save a recipe with a hotkey or to list all the recipes needed to craft a particular item. I don't know if there is a way to do this but, additionally, it would be nice if this system was aware of items you have in your inventory so it only tells you what to craft based on if you have it in your possession. Please let me know if you are open to implementing this.

**Comment by Chicken-Bones** [comment]
I've thought of an addon like this for a long time, but never had the time to make it, sorry.

**Comment by chozandrias76** [comment]
Yeah, I was looking into making an addon but I figured I had to learn quite a bit about java and your API before tackling it. I have a build environment set up but, haven't gone any farther.

**Comment by Chicken-Bones** [comment]
simply bookmarking items (so they can be at the top of your item list) would be fairly easy, anything more serious is varying degrees of significant effort

**Comment by cabralRodrigo** [comment]
Maybe something like another key binding to show all steps that NEI knows how to make an item and show like the current differents 'types of crafting' (Shaped, shapeless, smelting) and the steps could be a simple GUI with pagination on the bottom (like current multiples recipes is shown)

**Comment by Chicken-Bones** [comment]
if there are multiple ways to make an item, how will it know which one you want to use?

**Comment by cabralRodrigo** [comment]
I really don't know if this is good, but, my suggestion is using pagination (again) but this time on the top. Following the same behavior when I'll look an item and has more the one way to do in multiples 'types of crafting'

**Comment by cabralRodrigo** [comment]
This way the GUI is pretty much the same that is today.

**Comment by chozandrias76** [comment]
Chicken-Bones is there any way you could implement some sort of recipe saving or recently accessed recipes? It might help speed up people's workflow a TON when moving around their base and crafting.
